### PR TITLE
fix(Scripts/TheEye): Update Astromancer timings and add phase

### PR DESCRIPTION
**Pull request by balleny**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/6876
-  https://github.com/chromiecraft/chromiecraft/issues/6870
- https://github.com/chromiecraft/chromiecraft/issues/6958

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**
see sources in issues
## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by elthehablo** [comment]
Are the sources purely based on DBM timers? What are the sources on that? Or is that unconfirmed?

**Comment by balleny** [comment]
based on DBM timers, matching with https://youtu.be/FEYAX7l5H6A?t=5229
off by 0,8 seconds on this https://youtu.be/YlCfJB7RN6U?t=697
overall issue is that the whole transition isnt consistent with retail: she should teleport into the center first, then spawn 3 lightspots, then yell, then wait 1 second to disappear. Only then the 4 second timer for agents and the 20 seconds timer for becoming visible again should start. that is why many addons have 4 or 6 seconds timer. The yell is triggering the raid addons but she is actually just starting the timers after disappearing

so far my conclusion from observing different retail videos.

**Comment by elthehablo** [comment]
Okay that is fair. Please fix your titles for archiving purposes though

**Comment by balleny** [comment]
might actually make more sense to implement 5 seconds for the moment, but all the wrong things described can be observed well in here https://youtu.be/Cu3tFi68vK4?t=1917

**Comment by balleny** [comment]
added all but the line to trigger the blink spell
should likely be serverside spell not a CLEU one
coordinates based on sniffs provided @ heyitsbench

File: src/server/scripts/Outland/TempestKeep/Eye/boss_astromancer.cpp [hunk 1]
```diff
@@ -146,11 +150,30 @@ struct boss_high_astromancer_solarian : public BossAI
         }).Schedule(52100ms, [this](TaskContext context)
         {
             me->SetReactState(REACT_PASSIVE);
-            Talk(SAY_SUMMON);
-            me->RemoveAllAuras();
-            me->SetModelVisible(false);
-            scheduler.DelayAll(21s);
-            scheduler.Schedule(6s, [this](TaskContext)
+            scheduler.DelayAll(22s);
+            // blink to room center in this line using SPELL_TELEPORT_START_POSITION and START_POSITION_X, START_POSITION_Y, START_POSITION_Z
```
[the remark this answers is not in the document]
Comment (heyitsbench) [reply src/server/scripts/Outland/TempestKeep/Eye/boss_astromancer.cpp:154]: Current suspect is [33244](https://wowgaming.altervista.org/aowow/?spell=33244) (thanks to jedbouk on Discord for finding it) as it is noted as used by Solarian on TBCC Wowhead, would likely need its destination position set in DB to be used however.

**Comment by PkllonG** [comment]
PR what status. I'm using it on my server.

File: src/server/scripts/Outland/TempestKeep/Eye/boss_astromancer.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (elthehablo) [reply src/server/scripts/Outland/TempestKeep/Eye/boss_astromancer.cpp:154]: then let's incorporate that in here

[the remark this answers is not in the document]
Comment (balleny) [reply src/server/scripts/Outland/TempestKeep/Eye/boss_astromancer.cpp:154]: see also L41, its already included - just need someone to write the actual cast of the ability

[the remark this answers is not in the document]
Comment (balleny) [reply src/server/scripts/Outland/TempestKeep/Eye/boss_astromancer.cpp:154]: positions for the cast are also included in L59-61

**Review by elthehablo — APPROVED** [review]

File: src/server/scripts/Outland/TempestKeep/Eye/boss_astromancer.cpp [hunk 2]
```diff
@@ -146,11 +150,30 @@ struct boss_high_astromancer_solarian : public BossAI
...
+                    if (i == 0)
+                    {
+                        me->SummonCreature(NPC_ASTROMANCER_SOLARIAN_SPOTLIGHT, CENTER_X + cos(o)*INNER_PORTAL_RADIUS, CENTER_Y + std::sin(o)*INNER_PORTAL_RADIUS, CENTER_Z, 0.0f, TEMPSUMMON_TIMED_DESPAWN, 25000);
+                    }
+                    else
+                    {
+                        me->SummonCreature(NPC_ASTROMANCER_SOLARIAN_SPOTLIGHT, CENTER_X + cos(o)*OUTER_PORTAL_RADIUS, CENTER_Y + std::sin(o)*OUTER_PORTAL_RADIUS, PORTAL_Z, 0.0f, TEMPSUMMON_TIMED_DESPAWN, 25000);
+                    }
+                }
+            }).scheduler.Schedule(2s, [this](TaskContext)
```
Comment (elthehablo) [inline src/server/scripts/Outland/TempestKeep/Eye/boss_astromancer.cpp:169]: don't need to repeat .scheduler, you are still invoking Schedule on the same scheduler

File: src/server/scripts/Outland/TempestKeep/Eye/boss_astromancer.cpp [hunk 3]
```diff
@@ -146,11 +150,30 @@ struct boss_high_astromancer_solarian : public BossAI
...
+            {
+                Talk(SAY_SUMMON);
+            }).scheduler.Schedule(3s, [this](TaskContext)
```
Comment (elthehablo) [inline src/server/scripts/Outland/TempestKeep/Eye/boss_astromancer.cpp:172]: same here

**Review by elthehablo — CHANGES_REQUESTED** [review]

**Review by balleny — COMMENTED** [review]
done

**Review by elthehablo — APPROVED** [review]

**Comment by balleny** [comment]
This shouldnt have been merged yet given that that the teleport/blink spell was still missing
LL154
// blink to room center in this line using SPELL_TELEPORT_START_POSITION and START_POSITION_X, START_POSITION_Y, START_POSITION_Z

**Comment by elthehablo** [comment]
Nobody was gonna take this branch and write the script for you. So it should be incorporated in a different PR. Is the fight still working like this?

**Comment by elthehablo** [comment]
Looks that way but am on mobile so can't really check it well rn

**Pull request merged**